pkg/proxy/nftables/proxier.go

```diff
@@ -425,13 +434,16 @@ func ensureChain(chain string, tx *knftables.Transaction, createdChains sets.Set
 	if createdChains.Has(chain) {
 		return
 	}
+	createdChains.Insert(chain)
+	if tx == nil {
+		return
+	}
```

Comment:
should this not be checked first?

Reply:
we use `createdChains` to cleanup stale chains in the end of the iteration, so even if we skip adding given chain to the transaction (tx=nil), we still need to add it to the `createdChains` to not delete this valid chain in the end.

Reply:
Then please add a comment explaining that